app/helpers/fhir_helper.rb

```diff
@@ -886,6 +866,73 @@ def from_identifier(identifier, system_id, base_path)
     { value: id&.value, path: "#{base_path}.identifier[#{identifier&.index(id)}].value" }
   end
 
+  def to_contact_point(value, sys, rank, type, international, alternate)
+    return nil if value.blank?
+
+    contact_point = FHIR::ContactPoint.new(value: value, system: sys, rank: rank)
+
+    contact_point.extension << to_string_extension(type, 'phone-type') if type.present?
+
+    contact_point.extension << to_bool_extension(true, 'international-telephone') if international
+
+    contact_point.extension << to_bool_extension(true, 'alternate-contact') if alternate
+
+    contact_point
+  end
+
+  def from_telecom(telecom)
+    contact = {
+      email: { value: nil, path: '' },
+      primary_telephone: { value: nil, path: '' },
+      secondary_telephone: { value: nil, path: '' },
+      primary_telephone_type: { value: nil, path: '' },
+      secondary_telephone_type: { value: nil, path: '' },
+      international_telephone: { value: nil, path: '' },
+      alternate_email: { value: nil, path: '' },
+      alternate_primary_telephone: { value: nil, path: '' },
+      alternate_secondary_telephone: { value: nil, path: '' },
+      alternate_primary_telephone_type: { value: nil, path: '' },
+      alternate_secondary_telephone_type: { value: nil, path: '' },
+      alternate_international_telephone: { value: nil, path: '' }
+    }
+    int_phone_extension = to_bool_extension(true, 'international-telephone')
+    alt_extension = to_bool_extension(true, 'alternate-contact')
+
+    telecom&.each_with_index do |t, i|
+      international = t.extension.include?(int_phone_extension)
+      alternate = t.extension.include?(alt_extension)
+      case t.system
+      when 'email'
+        if alternate && contact[:alternate_email][:value].nil?
```

Comment:
are the purpose of these `nil?` checks so that if there are multiple telecoms of the same type, the first one would not be overwritten by any subsequent/duplicate values?

Reply:
Yep, the purpose is to keep the first one!